### PR TITLE
(IMAGES-791) Resolve Win-10 Sysprep Issues

### PR DESCRIPTION
**Pull request by jcoconnor**
This is a reworking of the Sysprep handling to resolve windows-10 (app related) issues.
The sysprep is moved to the build stage which also means a reduced vmpooler image
boot time.

It also contains:
1. Updates Cygwin (IMAGES-821)
2. Updates Chrome and other win utils (IMAGES-663)
3. Fixes a stderr logging issue with the pswindowsupdate (IMAGES-823)
4. Disable IPV6 on adapters (IMAGES-822)
5. Stop BGInfo/Powershell console popup (IMAGES-800)

### CODE REVIEW

File: templates/win/common/scripts/common/windows-env.ps1 [hunk 1]
```diff
@@ -411,115 +411,82 @@ Function Remove-AppsPackages {
...
+
+			  $AppName = $_.Name
+			  $AppFullName = $_.PackageFullName
+        Write-Output "Trying to remove $AppName ($AppFullName)"
+        
+        # Note - need to encase package removals in try catch to avoid loop fallout
+        # For some reason, the SilentlyContinue doesn't always appear to be honoured.
+			  try {
+          Write-Output "Removing $AppName for User"
+          Remove-AppxPackage -Package $AppFullName -ErrorAction SilentlyContinue
```
Comment (michaeltlombardi) [inline templates/win/common/scripts/common/windows-env.ps1]: Remove first for current user, then for all users?

File: templates/win/common/scripts/common/init-sysprep.ps1 [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    
+    Write-Host "Syspep fix for WMF5/Windows 2008R2"
+    reg.exe ADD "HKLM\SOFTWARE\Microsoft\Windows\StreamProvider"    /v "LastFullPayloadTime" /t REG_DWORD /d 0 /f
+}
+
+# Run the Application Package Cleaner again if required.
+if (Test-Path "$PackerLogs\AppsPackageRemove.Required") {
+    # Stop the tilemode service
+    net stop tiledatamodelsvc
+    Write-Output "Running Apps Package Cleaner post windows update"
```
Comment (michaeltlombardi) [inline templates/win/common/scripts/common/init-sysprep.ps1]: Why switching between `Write-Output` and `Write-Host`?

Comment (jcoconnor) [reply templates/win/common/scripts/common/init-sysprep.ps1]: Should be write-output across script - fixed.

File: templates/win/common/scripts/common/windows-env.ps1 [hunk 3]
```diff
@@ -411,115 +411,82 @@ Function Remove-AppsPackages {
...
     )
 
-    foreach ($app in $apps) {
-
-      Write-Output "Trying to remove $app"
-      try {
-        Get-AppxPackage -Name $app -AllUsers | Remove-AppxPackage -ErrorAction SilentlyContinue
-        Get-AppxPackage -Name $app -AllUsers | Remove-AppxPackage -AllUsers -ErrorAction SilentlyContinue
+	  Get-AppXPackage -Allusers |
+	  	Where-Object {$_.Name -notmatch ($KeepAppList -join '|') -and -not $_.IsFramework} |
```
[the remark this answers is not in the document]
Comment (jcoconnor) [reply templates/win/common/scripts/common/windows-env.ps1]: Two reasons:
1. Allows me to globally check/match for things like .Net using "Microsoft\.Net\..*" (-like would do this too).
2. Allows me to check for GUID type entries which also tend to be microsofty and not removable.

Aware of the performance penalty but thought it was a more useful way of doing the checks.

File: templates/win/common/scripts/common/windows-env.ps1 (hunk 1 above)
Comment (jcoconnor) [reply templates/win/common/scripts/common/windows-env.ps1]: Yes - I'm trying to recall why I did it that way - there was a reason.
Actually -AllUsers may be sufficient - there was a lot to and fro with this one.

Comment (jcoconnor) [reply templates/win/common/scripts/common/windows-env.ps1]: ```-Allusers``` looks to be sufficient, so taking the duplication out of here.

File: templates/win/common/scripts/common/windows-env.ps1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (michaeltlombardi) [reply templates/win/common/scripts/common/windows-env.ps1]: Okay!

**Comment by jcoconnor** [comment]
This PR is ready for Merge.
I have been able to build all the VMWare/vSphere builds successfully.
There are some issues with acceptance tests possibly requiring additional rework in ci-job-configs.

File: templates/win/common/scripts/common/sched-bginfo.vbs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+' It seems amazing that we still need to use VBS For some things in the year 2018 AD
+' However, Powershell insists on popping up a window (which grabs focus) every time the
+' bginfo scheduled task runs
```
Comment (michaeltlombardi) [inline templates/win/common/scripts/common/sched-bginfo.vbs]: Sounds like a good reason to drop bginfo...

Comment (jcoconnor) [reply templates/win/common/scripts/common/sched-bginfo.vbs]: Honest sir - all I was trying to do was to get an updating vmpooler machine lifetime for @Iristyle 
I'm happy to remove it or the updating if it causes too much difficulty.

**Pull request merged**